runtime/lua/vim/treesitter/query.lua

```diff
@@ -34,14 +72,45 @@ local function safe_read(filename, read_quantifier)
   return content
 end
 
+-- Disables loading query files for {lang} matching provided {patterns}.
+--
+-- If {patterns} is nil, it clears the ignore patterns.
+-- If {patterns} is string[], it replaces the currently set patterns (if any).
+--
+-- You can also pass a function as {patterns}, which will accept a table with
+-- currently set patterns as keys. You can modify that table and return a new
+-- table or nil. The return value will be used as the new ignore patterns.
+--
+---@param lang string
+---@param patterns nil | string[] | fun(patterns:table<string, boolean | nil>): nil | table<string, boolean | nil>
```

Comment:
We use `---` instead of `--` for doc comments:
```suggestion
--- Disables loading query files for {lang} matching provided {patterns}.
---
--- If {patterns} is nil, it clears the ignore patterns.
--- If {patterns} is string[], it replaces the currently set patterns (if any).
---
--- You can also pass a function as {patterns}, which will accept a table with
--- currently set patterns as keys. You can modify that table and return a new
--- table or nil. The return value will be used as the new ignore patterns.
---
---@param lang string
---@param patterns nil | string[] | fun(patterns:table<string, boolean | nil>): nil | table<string, boolean | nil>
```